src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -0,0 +1,822 @@
+        for (MentionType type : types)
+        {
+            switch (type)
+            {
+                case HERE:
+                {
+                    if (isMass("@here"))
+                        return true;
+                    break;
+                }
+                case EVERYONE:
+                {
+                    if (isMass("@everyone"))
+                        return true;
+                    break;
+                }
+                case USER:
+                {
+                    if (isUserMentioned(mentionable))
+                        return true;
+                    break;
+                }
+                case ROLE:
+                {
+                    if (isRoleMentioning(mentionable))
+                        return true;
+                    break;
+                }
+                case CHANNEL:
+                {
+                    if (mentionable instanceof TextChannel)
+                    {
+                        if (getMentionedChannels().contains(mentionable))
+                            return true;
+                    }
+                    break;
+                }
+                case EMOTE:
+                {
+                    if (mentionable instanceof Emote)
+                    {
+                        if (getEmotes().contains(mentionable))
+                            return true;
+                    }
+                    break;
+                }
+//              default: continue;
+            }
+        }
+        return false;
+    }
+
+    private boolean isUserMentioned(IMentionable mentionable)
+    {
+        if (mentionable instanceof User)
+        {
+            if (getMentionedUsers().contains(mentionable))
+                return true;
+        }
+        else if (mentionable instanceof Member)
+        {
+            final Member member = (Member) mentionable;
+            if (getMentionedUsers().contains(member.getUser()))
+                return true;
+        }
+        return false;
+    }
+
+    private boolean isRoleMentioning(IMentionable mentionable)
+    {
+        if (mentionable instanceof Role)
+        {
+            if (getMentionedRoles().contains(mentionable))
+                return true;
+        }
+        else if (mentionable instanceof Member)
+        {
+            final Member member = (Member) mentionable;
+            if (CollectionUtils.containsAny(getMentionedRoles(), member.getRoles()))
+                return true;
+        }
+        else if (isFromType(ChannelType.TEXT) && mentionable instanceof User)
+        {
+            final Member member = getGuild().getMember((User) mentionable);
+            if (CollectionUtils.containsAny(getMentionedRoles(), member.getRoles()))
+                return true;
+        }
+        return false;
+    }
+
+    private boolean isMass(String s)
+    {
+        return hasPermission(Permission.MESSAGE_MENTION_EVERYONE) && content.contains(s);
```

Comment:
This should use the `mentionsEveryone` fields as this can give incorrect results if the permissions of the author have been changed since the message has been sent.